### PR TITLE
Netflix/vmaf/#286

### PR DESCRIPTION
**Pull request by fishjam**
refactor for provide Result in libvmaf.h

progromer can get vmaf score from Result and output in speical format.
Usage:
```
std::unique_ptr<IVmafQualityRunner> spVmafQualityRunner =
  VmafQualityRunnerFactory::createVmafQualityRunner(...);
Asset asset(width, height, "yuv420p");
Result result = spVmafQualityRunner->run(asset, ...);
StatVector vmafResult = result.get_scores("vmaf");
```

### CODE REVIEW

**Comment by fishjam** [comment]
@li-zhi 
It's very strange that my change for c/c++ result in python UT fail, and I'm  unfamiliar with python and can not find the reason in a short time. Could you please help me to confirm and solve this PR?
Thank you.

**Comment by li-zhi** [comment]
Thanks! I'll probably look into this in the next two weeks.

**Comment by fishjam** [comment]
After more python study and test, I think the reason is python package conflict.
I build the docker and run the test command in [.travis.yml](https://github.com/Netflix/vmaf/blob/master/.travis.yml), there is same issue.

After I upgrade dask(1.0.0 => 1.1.0) and scikit-image(0.14.1 => 0.14.2), the issue has fixed.
Ref [attributeerror-numpy-ufunc-object-has-no-attribute-module](https://stackoverflow.com/questions/54192003/attributeerror-numpy-ufunc-object-has-no-attribute-module)
```
pip install --upgrade dask
pip install --upgrade scikit-image
```

BTW:
There are other issue remains, I'm not sure the reason and fix method(change code or change ut or others?)
```
.fopen ref_path /vmaf/python/test/resource/yuv/src01_hrc01_576x324_XXX.yuv failed.
.Start calculating VMAF score...
Input model at /vmaf/model/other_models/vmaf_v0.6.0.pkl_XXX cannot be read successfully.
Caught VmafException: Error loading model (.pkl): Trouble reading the file:/vmaf/model/other_models/vmaf_v0.6.0.pkl_XXX
.fopen ref_path /vmaf/python/test/resource/yuv/src01_hrc00_576x324_XXX.yuv failed.
.(width * height) % 2 != 0, width = 575, height = 323.
.Error: log_fmt must be xml or json, but is xml_XXX
.Error: pool_method must be min, harmonic_mean or mean, but is mean_XXX
.unknown format yuv420p_XXX.
.Start calculating VMAF score...
Caught logic_error: array: attempting to access element:18446744073709551613 in array of length:0
.Error: Invalid n_subsample value: 0

...

======================================================================
FAIL: test_run_ensemblevmaf_runner_different_models (quality_runner_test.QualityRunnerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vmaf/python/test/quality_runner_test.py", line 458, in test_run_ensemblevmaf_runner_different_models
    self.assertAlmostEqual(results[0]['EnsembleVMAF_model_1_score'], 81.78388541156974, places=4)
AssertionError: 81.78394259188538 != 81.78388541156974 within 4 places

======================================================================
FAIL: test_run_vmaf_runner_pool_perc10 (quality_runner_test.QualityRunnerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vmaf/python/test/quality_runner_test.py", line 626, in test_run_vmaf_runner_pool_perc10
    self.assertAlmostEqual(results[0]['VMAF_score'], 72.753572476835515, places=4)
AssertionError: 72.75385296763437 != 72.75357247683552 within 4 places

======================================================================
FAIL: test_run_vmafossexec_runner_enable_avx_precise (vmafossexec_test.VmafossexecQualityRunnerTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vmaf/python/test/vmafossexec_test.py", line 453, in test_run_vmafossexec_runner_enable_avx_precise
    self.assertAlmostEqual(results[0]['VMAFOSSEXEC_score'], 76.699266666666674, places=5)
AssertionError: 76.69930416666666 != 76.69926666666667 within 5 places

----------------------------------------------------------------------
Ran 274 tests in 1212.885s

FAILED (failures=3, skipped=1)

```

**Comment by fishjam** [comment]
Extra information about ci build fail.
  - 1.NumPy release 1.16.0 on 2019-01-14 02:39 UTC. [numpy](https://pypi.org/project/numpy/#history) 
  - 2.There is compability issue with dask, [dask#4378](https://github.com/dask/dask/issues/4378)
  - 3.I create PR and active the ci build on 2019-01-14T10:27:36Z

Then build fail :-(

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ before_install:
 install:
   - pip install --upgrade pip
   - pip install --upgrade numpy scipy matplotlib pandas
+  - pip install --upgrade dask
```
Comment (li-zhi) [inline .travis.yml]: Is this required?

**Comment by li-zhi** [comment]
@fishjam thanks for the effort! There will be another PR to merge with big changes to speed up VMAF. I'll merge that one first before yours.

File: .travis.yml (hunk 1 above)
Comment (fishjam) [reply .travis.yml]: according https://github.com/Netflix/vmaf/pull/295 's result, it does not need.
it seems that current pip package repository has the right version, dask(1.1.0) and scikit-image(0.14.2). 
But I need do something to active the ci build, so ...

Comment (li-zhi) [reply .travis.yml]: Ok. It sounds non-essential to the vmaf package. Would you mind removing this?

Comment (fishjam) [reply .travis.yml]: I reset the commitid to e1c4458(fix bug: crash while predictionStructs empty), the last commit that I want, and push --force,  but it seems that CI can not be re-activate to re-check it.
Do I have to commit some unwanted?

Comment (li-zhi) [reply .travis.yml]: Could you forward to the latest commit, and add a new commit just to remove that line?

Comment (fishjam) [reply .travis.yml]: format code  to same checkstyle and activate the CI build,

**Pull request merged**